### PR TITLE
[Refactor][BugFix] P-eagle and lmhead bugfix and Reduce sample reactor

### PR DESCRIPTION
**Pull request by lilinsiman**
### What this PR does / why we need it?
When both the P-eagle and lmhead features are enabled, dummyrun is used for accompaniment, which causes token_indices_to_sample to be too long and exceed the value of max_num_reqs_across_dp. As a result, in lmhead, no padding is performed. Therefore, when P-eagle is enabled, an error is reported at position 1 exit. The solution is to pad the corresponding length to the initial length and restore the length when num_indices is too long.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
UT and Tests

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/967c5c3bc38891f4465d3f4e99917ed837bb3833

### CODE REVIEW

**Review by slippersss — APPROVED** [review]

File: vllm_ascend/spec_decode/llm_base_proposer.py [hunk 1]
```diff
@@ -976,15 +976,19 @@ def _propose(
         return draft_token_ids
 
     def compute_draft_token_ids(self, hidden_states: torch.Tensor):
-        logits = self.model.logits_processor(self.model.lm_head, hidden_states)
-        if not hasattr(self.model, "draft_id_to_target_id") or self.model.draft_id_to_target_id is None:
+        if self.method in ("eagle3", "dflash"):
```
Comment (winson-00178005) [inline vllm_ascend/spec_decode/llm_base_proposer.py:979]: ## `compute_draft_token_ids` 分支逻辑建议

```python
if self.method in ("eagle3", "dflash"):
    logits = self.model.logits_processor(self.model.lm_head, hidden_states)
    ...
else:
    logits = self.model.compute_logits(hidden_states)
    return greedy_sample(logits)
```

`compute_draft_token_ids` 根据 `self.method` 选择不同的 logits 计算路径，但 `eagle3/dflash` 走 `logits_processor(lm_head)`，其他走 `compute_logits`。这两条路径的语义差异（是否经过 quant 方法等）没有在代码或注释中说明。建议补充说明为什么 eagle3/dflash 需要走 `logits_processor` 而不是 `compute_logits`，以及两者在量化场景下的行为差异。

**Review by zzzzwwjj — APPROVED** [review]
